packages/core/src/Stores/common-store.js

```diff
@@ -148,6 +148,7 @@ export default class CommonStore extends BaseStore {
             this.app_routing_history.splice(check_existing, 1);
         }
         this.app_routing_history.unshift(router_action);
+        this.app_routing_history = this.app_routing_history.slice();
```

Comment:
why do we add this? this is a potential breaking change for users of `app_routing_history`

Reply:
ah, my mistake. I should slice it on `autorun` function